### PR TITLE
Bug fix: when registry is unreachable during re-register process, ser…

### PR DESCRIPTION
**Pull request by humingcheng**
when registry is unreachable during re-register process, serviceId may be set to empty and could never be revoverd

### CODE REVIEW

File: core/registry/bootstrap.go [hunk 1]
```diff
@@ -52,11 +53,11 @@ func RegisterMicroservice() error {
 	lager.Logger.Infof("Micro service registered by [ %s ]", framework.Register)
 
 	sid, err := DefaultRegistrator.RegisterService(microservice)
-	runtime.ServiceID = sid
```
Comment (tianxiaoliang) [inline core/registry/bootstrap.go]: if sid !="" 
you mist set it to runtime

Comment (tianxiaoliang) [reply core/registry/bootstrap.go]: assume it is the first time to register service, then runtime sid will be empty

File: core/registry/bootstrap.go [hunk 2]
```diff
@@ -52,11 +53,11 @@ func RegisterMicroservice() error {
...
 	if err != nil {
 		lager.Logger.Errorf(err, "Register [%s] failed", microservice.ServiceName)
 		return err
 	}
+	runtime.ServiceID = sid
```
Comment (tianxiaoliang) [inline core/registry/bootstrap.go]: add one more check if sid!="" else give error log

Comment (humingcheng) [reply core/registry/bootstrap.go]: Done

**Pull request merged**